include/zephyr/llext/symbol.h

```diff
@@ -117,18 +117,22 @@ struct llext_symtable {
  *
  * This macro can be used in extensions to add a symbol (function or object)
  * to the extension's exported symbol table, so that it may be referenced by
- * the base image.
+ * the base image or by other extensions.
  *
  * @param x Extension symbol to export to the base image
  */
-#if defined(CONFIG_LLEXT) && defined(LL_EXTENSION_BUILD)
+#if defined(CONFIG_LLEXT)
+#if defined(LL_EXTENSION_BUILD)
```

Comment:
I think the rationale behind this is to allow exporting symbols from code that may be built in-application or as an LLEXT (as done in `samples/subsys/llext/modules`): https://github.com/zephyrproject-rtos/zephyr/blob/143b14bb4fb779437fdeb8369e42fbb81db4a5fd/samples/subsys/llext/modules/src/hello_world_ext.c#L17-L27

IMO, we should rename `EXPORT_SYMBOL` to `EXPORT_BUILTIN_SYMBOL`, and rename the existing `LL_EXTENSION_SYMBOL` to `EXPORT_EXTENSION_SYMBOL`; then, we would have a single e.g. `LLEXT_SYMBOL_EXPORT` macro that would be used everywhere and could select between the two depending on the type of build.
(For Zephyr code, this would always turn into `EXPORT_BUILTIN_SYMBOL`, unless someone starts having fun with this to make the kernel modular :p)

```c
#if defined(CONFIG_LLEXT)
#define EXPORT_BUILTIN_SYMBOL(x)  ... /* this should select between SLID and non-SLID but you get the idea */
#define EXPORT_EXTENSION_SYMBOL(x) ... /* current LL_EXTENSION_SYMBOL here */
#else
/* stub defines */
#define EXPORT_BUILTIN_SYMBOL(x)
#define EXPORT_EXTENSION_SYMBOL(x)
#endif

#if defined(LL_EXTENSION_BUILD)
#define LLEXT_SYMBOL_EXPORT(x) EXPORT_EXTENSION_SYMBOL(x)
#else
#define LLEXT_SYMBOL_EXPORT(x) EXPORT_BUILTIN_SYMBOL(x)
#endif
```

(sidenote: LL_EXTENSION_BUILD is lacking documentation, although IDK where this kind of macro should be documented)

Reply:
The thing is... EXPORT_SYMBOL can be defined to be whatever it needs to be.

E.g. when building an extension it can be defined to do something different than when being built in the base firmware if needed. I don't think we need two distinct macros here myself.

Reply:
Maybe I worded it poorly, but this is exactly what I wanted to propose: have a single macro used to export symbols, regardless of where we are exporting from, that would automagically select the proper implementation (i.e., `STRUCT_SECTION_ITERABLE(llext_const_symbol, x ## _sym)` if building in base firmware, `Z_GENERIC_SECTION(".exported_sym")` if building in extension).

Maybe I should have used the `Z_` prefix to make it clear that `EXPORT_BUILTIN_SYMBOLS`/`EXPORT_EXTENSION_SYMBOL` are internal-usage-only macros in my proposal snippet.

Reply:
@teburd agree, just like in the Linux kernel